build/Build.cs

```diff
@@ -539,22 +547,34 @@ AbsolutePath DoPublish(string project, string framework, string version, string?
 
             return publishedTo;
         }
-
-        void SignAndPack(string project, DotNetPackSettings dotNetCorePackSettings)
+        
+        void SignProject(string project)
         {
+            if (!WillSignBinaries)
+                return;
             Log.Information("SignAndPack project: {Project}", project);
+            var binDirectory = $"{Path.GetDirectoryName(project)}/bin/{Configuration}/";
+            SignDirectory(binDirectory);
+        }
 
-            if (WillSignBinaries)
-            {
-                var binDirectory = $"{Path.GetDirectoryName(project)}/bin/{Configuration}/";
-                var binariesFolders =
-                    Directory.GetDirectories(binDirectory, "*", new EnumerationOptions { RecurseSubdirectories = true });
-                foreach (var directory in binariesFolders)
-                    Signing.SignAndTimestampBinaries(directory, AzureKeyVaultUrl, AzureKeyVaultAppId,
-                                                     AzureKeyVaultAppSecret, AzureKeyVaultTenantId, AzureKeyVaultCertificateName,
-                                                     SigningCertificatePath, SigningCertificatePassword);
-            }
+        void SignDirectory(string directory)
+        {
+            if (!WillSignBinaries)
+                return;
+            Signing.SignAndTimestampBinaries(directory, AzureKeyVaultUrl, AzureKeyVaultAppId,
+                                             AzureKeyVaultAppSecret, AzureKeyVaultTenantId, AzureKeyVaultCertificateName,
+                                             SigningCertificatePath, SigningCertificatePassword);
+            var binariesFolders =
+                Directory.GetDirectories(directory, "*", new EnumerationOptions { RecurseSubdirectories = true });
+            foreach (var subDirectory in binariesFolders)
+                Signing.SignAndTimestampBinaries(subDirectory, AzureKeyVaultUrl, AzureKeyVaultAppId,
+                                                 AzureKeyVaultAppSecret, AzureKeyVaultTenantId, AzureKeyVaultCertificateName,
+                                                 SigningCertificatePath, SigningCertificatePassword);
```

Comment:
Looks like this part can be simplified a bit:

```suggestion
            var binariesFolders =
                Directory.GetDirectories(directory, "*", new EnumerationOptions { RecurseSubdirectories = true });
            foreach (var dir in binariesFolders.Append(directory))
                Signing.SignAndTimestampBinaries(dir, AzureKeyVaultUrl, AzureKeyVaultAppId,
                                                 AzureKeyVaultAppSecret, AzureKeyVaultTenantId, AzureKeyVaultCertificateName,
                                                 SigningCertificatePath, SigningCertificatePassword);
```